packages/internal-test-env/index.ts

```diff
@@ -0,0 +1,351 @@
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+import { config } from "dotenv-flow";
+import { join } from "path";
+import { Session } from "@inrupt/solid-client-authn-node";
+import {
+  createContainerInContainer,
+  createSolidDataset,
+  deleteSolidDataset,
+  getPodUrlAll,
+  getSourceIri,
+  saveSolidDatasetInContainer,
+} from "@inrupt/solid-client";
+
+const availableEnvironment = [
+  "ESS Dev-Next" as const,
+  "ESS PodSpaces" as const,
+  "NSS" as const,
+  //  "ESS PodSpaces Next" as const,
+];
+
+export type AvailableEnvironment = typeof availableEnvironment extends Array<
+  infer E
+>
+  ? E
+  : never;
+
+const availableProtocol = ["ESS Notifications Protocol" as const];
+
+export type AvailableProtocol = typeof availableProtocol extends Array<infer E>
+  ? E
+  : never;
+
+export interface TestingEnvironmentNode {
+  environment: AvailableEnvironment;
+  idp: string;
+  notificationGateway: string | undefined;
+  protocol: AvailableProtocol | undefined;
+  clientCredentials: {
+    requestor: {
+      id: string;
+      secret: string;
+    };
+    resourceOwner: {
+      id: string | undefined;
+      secret: string | undefined;
+    };
+  };
+  vcProvider: string | undefined;
+  features: FeatureFlags | undefined;
+}
+export interface TestingEnvironmentBrowser {
+  clientCredentials: {
+    resourceOwner: {
+      login: string;
+      password: string;
+    };
+  };
+  idp: string;
+  notificationGateway: string | undefined;
+  features: FeatureFlags | undefined;
+}
+
+type FeatureFlags = {
+  [key: string]: any;
+};
+
+export interface EnvVariables {
+  // Common Envs
+  E2E_TEST_ENVIRONMENT: AvailableEnvironment;
+  E2E_TEST_IDP: string;
+  E2E_TEST_USER: string;
+  E2E_TEST_PASSWORD: string;
+
+  // Client credentials for the access requestor
+  E2E_TEST_REQUESTOR_CLIENT_ID: string;
+  E2E_TEST_REQUESTOR_CLIENT_SECRET: string;
+
+  // Needed for solid-notifications-js
+  E2E_TEST_NOTIFICATION_GATEWAY: string | undefined;
+  E2E_TEST_NOTIFICATION_PROTOCOL: AvailableProtocol | undefined;
+
+  // VC service provider
+  E2E_TEST_VC_PROVIDER: string | undefined;
+
+  // Client credentials for the resource owner
+  E2E_TEST_RESPONDER_CLIENT_ID: string | undefined;
+  E2E_TEST_RESPONDER_CLIENT_SECRET: string | undefined;
```

Comment:
This looks a bit odd? `RESPONDER`? Maybe `OWNER` or `CONTROLLER` ? Comment also doesn't match